### PR TITLE
Fix #1972

### LINKED ISSUE #1972: Extend VertxLifecycleHooks to support shutdown
The VertxLifecycleHooks class does not notify the `Launcher` when the `vertx` instance is shutdown. We should add (ideas):

* `beforeStoppingVertx` - called during the `close` method of vert.x (CloseHook)
* `afterStoppingVertx` - called once vert.x has been closed

### PR DESCRIPTION
**Pull request by cescoffier**
Add:
* beforeStoppingVertx - called during the close method of vert.x (CloseHook)
* afterStoppingVertx - called once vert.x has been closed

### CODE REVIEW

**Comment by cescoffier** [comment]
@tsegismont that's the PR to add shutdown hooks (discussed on the mailing list). Can you review?

(related issue if #1972)

File: src/main/java/io/vertx/core/impl/launcher/commands/BareCommand.java [hunk 1]
```diff
@@ -298,12 +311,29 @@ private Method getSetter(String fieldName, Class<?> clazz) {
...
+   * @param action the action, may be {@code null}
    */
-  protected void addShutdownHook() {
-    Runtime.getRuntime().addShutdownHook(new Thread() {
-      public void run() {
-        CountDownLatch latch = new CountDownLatch(1);
+  public static Runnable getTerminationRunnable(Vertx vertx, Logger log, Runnable action) {
+    return () -> {
+      CountDownLatch latch = new CountDownLatch(1);
+      System.out.println("Shutdown hook ! " + vertx);
```
Comment (tsegismont) [inline src/main/java/io/vertx/core/impl/launcher/commands/BareCommand.java]: Can you remove this? I guess it was for debugging purpose?

File: src/test/java/io/vertx/core/impl/launcher/LauncherExtensibilityTest.java [hunk 2]
```diff
@@ -111,7 +115,11 @@ protected void load() {
   }
 
   @Test
-  public void testThatCustomLauncherCanCustomizeTheClusteredOption() {
+  public void testThatCustomLauncherCanCustomizeTheClusteredOption() throws InterruptedException {
+
+    AtomicBoolean asv = new AtomicBoolean();
+    AtomicBoolean bsv = new AtomicBoolean();
```
Comment (tsegismont) [inline src/test/java/io/vertx/core/impl/launcher/LauncherExtensibilityTest.java]: Where is bsv tested? It seems it's only updated.

**Comment by cescoffier** [comment]
@tsegismont I've fixed the two issues.

**Comment by tsegismont** [comment]
LGTM

**Pull request merged**